src/cpu/pred/btb/decoupled_bpred.cc

```
@@ -132,44 +132,51 @@ DecoupledBPUWithBTB::tick()
-    if (bpuState == BpuState::IDLE && !targetQueueFull()) {
-        if (blockPredictionPending) {
-            DPRINTF(Override, "Prediction blocked to prioritize resolve update\n");
-            dbpBtbStats.predictionBlockedForUpdate++;
-            blockPredictionPending = false;
-        } else {
-            requestNewPrediction();
-            bpuState = BpuState::PREDICTOR_DONE;
+    int predsRemainsToBeMade = enableTwoTaken ? 2 : 1;
+    unsigned tempNumOverrideBubbles = 0;
+
+    while (predsRemainsToBeMade > 0) {
```

Cppcheck (2.19.0)

[error] Shifting 64-bit value by 64 bits is undefined behaviour

(shiftTooManyBits)